### PR TITLE
Added update percentage map, to allow for a percentage of a node pool

### PR DESCRIPTION
**Pull request by thorfour**
New config for TNC to communicate the percentage of nodes that are allowed to update at a time for a given role.

### CODE REVIEW

File: config/tectonic-node-controller/config.go [hunk 1]
```diff
@@ -32,6 +32,9 @@ type ControllerConfig struct {
 
 	// User customizations, list of node configs in the cluster to collect and apply as part of the final config
 	AdditionalConfigs []string `json:"additionalConfigs"`
+
+	// Pool update percentages allowing individual roles to have a specific percentage of nodes that can be upgrading at a given time ie. worker=10%, if 0% only a single node will be updated at a given time
+	NodePoolUpdatePercent map[string]int
```
Comment (yifan-gu) [inline config/tectonic-node-controller/config.go]: json tag?

Comment (yifan-gu) [reply config/tectonic-node-controller/config.go]: @aaronlevy OK to include this knob?

Comment (aaronlevy) [reply config/tectonic-node-controller/config.go]: Yeah, I think it's reasonable - however, I think we need to allow absolute as well as percentages. Reason for this are things like etcd (only 1 node at a time). Maybe we do this similar to deployments `maxUnavailable` (which can be percent or absolute number). See: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#strategy

Thoughts?

Comment (yifan-gu) [reply config/tectonic-node-controller/config.go]: Sounds reasonable, then we need to change this to a string, the format will be either integer (e.g. `1`) or percentage (e.g. `10%`). Maybe also rename it to `NodePoolUpdateLimit` ? @thorfour

Comment (thorfour) [reply config/tectonic-node-controller/config.go]: That sounds reasonable

**Pull request merged**